protocol_grpc.go

```diff
@@ -325,8 +325,6 @@ func (g *grpcClient) NewConn(
 		}
 	} else {
 		conn.readTrailers = func(_ *grpcUnmarshaler, call *duplexHTTPCall) http.Header {
-			// To access HTTP trailers, we need to read the body to EOF.
-			_ = discard(call)
 			return call.ResponseTrailer()
 		}
```

Comment:
I'm not sure that it's safe to remove this: if the client is using the gRPC protocol, the code you've added to `envelope.go` won't run (because end-of-stream data isn't sent in the body). I think we'll want the `readTrailers` function to also return an error, so we can error out here if the client hasn't read the whole stream or there are I/O errors.

Should all become clearer once you've got tests in place though :)

Reply:
Reverted, hard to write a test case for it so removed from the excess trailing data test case.